fix(NcDateTimePicker): change styling to look more like the native picker

GretaD opened the pull request:
### ☑️ Resolves

- Fix # https://github.com/nextcloud/calendar/pull/6322

### 🖼️ Screenshots

🏚️ Before | 🏡 After
after
![Screenshot from 2024-09-30 11-52-42](https://github.com/user-attachments/assets/1d12da6e-f6e0-4754-b31b-ef7452c55735)




### 🚧 Tasks

- [ ] ...

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [ ] 📘 Component documentation has been extended, updated or is not applicable
- [ ] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

susnux commented on src/components/NcDateTimePicker/NcDateTimePicker.vue:
```diff
@@ -477,10 +477,53 @@ export default {
+
+.mx-datepicker-main .mx-table .cell {
+	border-radius: var(--border-radius-small);
+	min-height: 30px !important;
```
This should be probably either default clickable or clickable-small right?

GretaD replied:
yes, and also, i need to fix the classes to be shown as BEM.

susnux replied:
But that are internal classes, so no renaming to BEM possible, no?

GretaD replied:
i meant:

```
main {
.cell {
.today {
.other.. {
}
 }
  }
   }
```

nimishavijay reviewed (changes requested):
Looks really good! Very big improvement when we compare it to the current design :) 
- [ ] Is it possible to hide the first row in this case? The topmost week that is shown should be the week containing the 1st of the month (eg. we are on September but the last week of August is also shown)
- [x] Is it possible to hide the double arrows which skip to the same month next year (`«` and `»`) since I imagine that functionality is not used very often. This would also provide more horizontal space for the month name :) 
- [x] As @susnux mentioned the size should be a variable (`--clickable-area-small` should work here)
- [x] Nice-to-have: Show full month name

GretaD commented:
> Looks really good! Very big improvement when we compare it to the current design :)
> 
> * [ ]  Is it possible to hide the first row in this case? The topmost week that is shown should be the week containing the 1st of the month (eg. we are on September but the last week of August is also shown)

Unfortunately, i cannot find a way, but i removed the background color, so it doesnt look too crowded.

The rest is done. I updated to screenshot

susnux reviewed (changes requested):
Looks nice, **but** it is no longer possible to select a year that is more in the past / future as the buttons are removed:

before | now
---|---
![image](https://github.com/user-attachments/assets/ce3ed609-b38a-47db-894f-c31bb775c95b)|![image](https://github.com/user-attachments/assets/011f5e33-ecd6-484f-9355-9d4a87963460)


Also when selecting the month the year should be centered like before:

before | now
---|---
![image](https://github.com/user-attachments/assets/f6d5eaa6-f1e2-488d-a4b9-88f6f9e27e0a)|![image](https://github.com/user-attachments/assets/7684d520-9060-4269-bb87-9deb8523415a)

GretaD commented:
> Looks nice, **but** it is no longer possible to select a year that is more in the past / future as the buttons are removed:


ahhh thanks for testing it, @nimishavijay i guess we need the buttons :D

GretaD commented:
> Please also have a look at #6125. I missed that you were also working on the date picker design 🙈
> 
> Ref [nextcloud/calendar#6324](https://github.com/nextcloud/calendar/issues/6324)

we can merge them, no problem, the change is the same, with small difference, like the hover button state. Do you want to include mine at yours, or should i add yours here?
